django/utils/numberformat.py

```diff
@@ -22,12 +22,18 @@ def format(number, decimal_sep, decimal_pos=None, grouping=0, thousand_sep='',
     use_grouping = use_grouping or force_grouping
     use_grouping = use_grouping and grouping != 0
     # Make the common case fast
-    if isinstance(number, int) and not use_grouping and not decimal_pos:
-        return mark_safe(number)
-    # sign
-    sign = ''
-    if isinstance(number, Decimal):
+    if isinstance(number, int) and (not use_grouping or grouping == 3):
```

Comment:
You're repeating `not grouping`, I am sure there is a way to rearrange the if statements a bit.